patroni/ctl.py

```diff
@@ -1372,3 +1373,205 @@ def format_pg_version(version):
         return "{0}.{1}.{2}".format(version // 10000, version // 100 % 100, version % 100)
     else:
         return "{0}.{1}".format(version // 10000, version % 100)
+
+
+def enrich_config_from_running_instance(dsn: str, config: Dict[str, Any], no_value_msg: str) -> None:
+    from patroni.postgresql.config import parse_dsn
+    from patroni.config import _AUTH_ALLOWED_PARAMETERS
+
+    parsed_dsn = parse_dsn(dsn)
+    if not parsed_dsn:
+        raise PatroniCtlException('Failed to parse DSN string')
+    if not parsed_dsn.get('user'):
+        raise PatroniCtlException('No user provided in the DSN')
+
+    from . import psycopg
+    conn = psycopg.connect(dsn=dsn)
+
+    with conn.cursor() as cur:
+        cur.execute("SELECT 1 FROM pg_roles WHERE rolname=%s AND rolsuper='t';", (parsed_dsn['user'],))
+        if cur.rowcount < 1:
+            conn.close()
+            raise PatroniCtlException('The provided user does not have superuser privilege')
+
+        cur.execute("SELECT name, setting \
+                     from pg_settings \
+                     where setting <> reset_val and setting <> '(disabled)' \
+                     or name in ('port', 'listen_addresses');")
```

Comment:
There is a bunch of parameters that we want to read from pg_setting no matter whether they are default or not: https://github.com/zalando/patroni/blob/master/patroni/postgresql/config.py#L272-L283

Reply:
we want to read all of them even having all but `wal_level` and `wal_keep_segments`/`wal_keep_size` set to their PG defaults in the code you linked?

and I guess here we should also exclude all the settings that have `context='internal'` here